Fix fgets implementation again

shinh opened the pull request:
When FILE is already at EOF:

- if size<=0, ret=nullptr and buf[0] is not modified
- if size==1, ret=buf and buf[0]=0
- if size>1, ret=nullptr and buf[0] is not modified

EDIT: thanks to https://github.com/shinh/elvm/pull/99#issuecomment-778339579

umutoztunc commented:
I have tested the behaviour of `fgets` locally. Here are my results:

- if size <= 0, return null and buffer is not modified
- if size == 1, return buffer and buffer[0]=0
- if size > 1, return null and buffer is not modified

shinh commented:
Yeah, at first I removed 
```
  if (size <= 0)
    return NULL;	    return NULL;
```
but I realized removing this breaks the `size<=0` case. I reverted removing it but forgot to update the PR description. I think now this works for the all three cases. The test cases should cover them, too.
